### ISSUE TITLE
How to escape include

### ISSUE THREAD

**Issue by marek95** [mozilla/nunjucks#695]
Hi,
is it be possible to somehow escape HTML of included partial?
I tried this but there is no output

`
{% filter escape %}
    {% include "item.html" %}
{% endfilter %}
`

is there any way to do it? 
Thanks!

**Comment by carljm** [comment]
That is the way you should do it, but currently there is a bug with the `filter` tag not handling nested tags properly and truncating the content instead: #576

**Comment by marek95** [comment]
thanks! is there any way to get around it now?

**Comment by carljm** [comment]
Not any easy way that I know of :/ Block `set` (#656) would provide a workaround, but that's not released yet... I'll do a release right now.

**Comment by carljm** [comment]
I released 2.4.0, so you should be able to work around this with something like:

```
{% set itemHtml %}
  {% include "item.html" %}
{% endset %}
{{ itemHtml|escape }}
```

(though the use of `|escape` filter there may not even be necessary if autoescape is on)

**Comment by marek95** [comment]
thanks for fast reply!
it does not work tho... 
if i do this:
```
{% set itemHtml %}
  <p>hahaha</p>
{% endset %}
{{ itemHtml | escape }}
```
output is this (as expeced)
```
&amp;lt;p&amp;gt;hahaha&amp;lt;/p&amp;gt;
```
but with your example it does not work. btw i work with grunt plugin nunjucks-2-html and it does not throw any error... it just does not output anything

**Comment by carljm** [comment]
Sorry, I'm not sure what's going on then. Did you upgrade to the just-released nunjucks 2.4.0? If so, my example should work; we have tests for includes within a block-set. I haven't used the grunt plugin you're using.

**Comment by marek95** [comment]
yes, i have upgraded, i have nunjucks  2.4.0 in node_modules folder ... when i tried that code before it has shown some error, after upgrade it did not output anything 
 i am tired of this... i cant find a tool that allows to escape includes in html if i need to, i wouldn't have thought that it would be such a problem ...
... anyways thanks much for help!
